packages/volto/src/express-middleware/devproxy.js

```diff
@@ -47,10 +47,16 @@ export default function devProxyMiddleware() {
       },
     ),
     onProxyReq: (proxyReq, req, res) => {
+      //Fixes https://github.com/plone/volto/issues/6283
+      proxyReq.setHeader('Host', new URL(config.settings.apiPath).host);
       // Fixes https://github.com/chimurai/http-proxy-middleware/issues/320
       if (!req.body || !Object.keys(req.body).length) {
         return;
       }
+      // Set headers for the proxy request
+      proxyReq.setHeader('X-Real-IP', req.ip);
+      proxyReq.setHeader('X-Forwarded-For', req.ip);
```

Comment:
If the incoming request already has X-Forwarded-For from another proxy, we should keep that value and add req.ip to the end of the list: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For